mobile/lib/profile/screens/profile_page.dart

```diff
@@ -0,0 +1,217 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:learning_api/api.dart';
+import 'package:mobile/api/ApiClient.dart';
+import 'package:mobile/common/custom_navigation_bar.dart';
+import 'package:mobile/profile/widgets/custom_counter.dart';
+import 'package:mobile/profile/widgets/custom_counter_description.dart';
+
+class ProfilePage extends StatefulWidget {
+  const ProfilePage({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _ProfilePageState();
+}
+
+class _ProfilePageState extends State<ProfilePage> {
+  late Future<ProfilePageData> dataFuture;
+
+  @override
+  void initState() {
+    super.initState();
+
+    dataFuture = getUserProfileData();
+  }
+
+  Future<ProfilePageData> getUserProfileData() async {
+    ProfilePageData profilePageData = ProfilePageData();
+    await UserEndpointApi(apiClient).getCurrentUserProfile().then((value) => {
+          profilePageData.username = value?.username,
+          profilePageData.email = value?.email,
+          profilePageData.friendsCount = value?.friendsCount,
+          profilePageData.finishedActivitiesCount =
+              value?.finishedActivitiesCount,
+          // add daily streak data
+          // profilePageData.dailyStreak = value?.dailyStreak
+        });
+
+    return profilePageData;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final double screenWidth = MediaQuery.of(context).size.width;
+    final double screenHeight = MediaQuery.of(context).size.height;
+
+    return MaterialApp(
+        home: Scaffold(
+            body: SafeArea(
+      child: FutureBuilder<ProfilePageData?>(
+        future: dataFuture,
+        builder: (context, snapshot) {
+          if (snapshot.hasData) {
+            ProfilePageData profilePageData = snapshot.data!;
+            return Column(
+              children: [
+                Flexible(
+                  child: SingleChildScrollView(
+                    child: Column(
+                      children: [
+                        Padding(
+                          padding: EdgeInsets.only(top: screenHeight * 0.02),
+                          child: Image.asset('assets/images/user.png',
+                              width: 70, height: 70),
+                        ),
+                        Padding(
+                          padding: EdgeInsets.symmetric(
+                              vertical: screenHeight * 0.01),
+                          child: const Text(
+                            'Profil użytkownika',
+                            style: TextStyle(
+                                fontSize: 34,
+                                color: Colors.black,
+                                fontWeight: FontWeight.normal),
+                          ),
+                        ),
+                        Padding(
```

Comment:
to rozbić jako oddzielny widget? nie wiem ten kod się robi strasznie nieczytelny jak nie jest porozbijane :(